src/app/services/theme/theme.service.ts

```diff
@@ -102,6 +104,37 @@ export class ThemeService {
     }
   }
 
+  private overrideBootstrapColors(color: string) {
+    const variables: [string, number][] = [
+      ["--bs-link-color", 0],
+      ["--bs-link-hover-color", -0.1],
+      ["--bs-link-disabled-color", 0.1],
+    ];
+
+    const targetColor = hsl(color);
+
+    variables.forEach(([variable, lightnessAdjustment]) => {
+      const hslColor = targetColor.copy({ l: targetColor.l + lightnessAdjustment });
+      const rgbColor = rgb(hslColor);
+
+      // set the color in the browser
+      this.root.style.setProperty(variable, rgbColor.toString());
+      this.root.style.setProperty(
+        `${variable}-rgb`,
+        `${rgbColor.r}, ${rgbColor.g}, ${rgbColor.b}`
+      );
+
+      // set the color in SSR
+      const style = this.document.createElement("style");
+      style.innerHTML = `
+      :root {
+        ${variable}: ${rgbColor.toString()};
+        ${variable}-rgb: ${rgbColor.r}, ${rgbColor.g}, ${rgbColor.b};
+      }
+    `;
```

Comment:
ok ignore my special case comment from above

but we should split the SSR and the client versions into their own functions:

- for consistency (with the other pair of set color functions in this file)
- and to makes ure SSR is not attempting to access dom elements (this may not be an issue, but since I believe it results in no effect, we shouldn't just leave it in - it is confusing having code that does nothing and it is extra work)